perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -346,48 +355,105 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
     sizes: the object sizes used in the benchmark, in bytes.
     results: a list to append Sample objects to.
     metadata: dict. Base sample metadata
-
   """
 
+  num_streams = FLAGS.object_storage_multistream_num_streams
+
   if metadata is None:
     metadata = {}
-
-  records = json.loads(raw_result)
-  metric_name = 'Multi-stream %s latency' % operation
-
-  metadata = metadata.copy()
-  metadata['num_streams'] = FLAGS.object_storage_multistream_num_streams
+  metadata['num_streams'] = num_streams
   metadata['objects_per_stream'] = (
       FLAGS.object_storage_multistream_objects_per_stream)
 
-  overall_metadata = metadata.copy()
+  records_json = json.loads(raw_result)
+  records = pd.DataFrame(records_json)
+
+  all_streams_start, all_streams_duration = analyze.AllStreamsInterval(
+      records['start_time'], records['latency'], records['stream_num'])
+  start_gap, stop_gap = analyze.StreamStartAndEndGaps(
+      records['start_time'], records['latency'],
+      all_streams_start, all_streams_duration)
+  if ((start_gap + stop_gap) / all_streams_duration <
+      MULTISTREAM_STREAM_GAP_THRESHOLD):
+    logging.info(
+        'First stream started %s seconds before last stream started', start_gap)
+    logging.info(
+        'Last stream ended %s seconds after first stream ended', stop_gap)
+  else:
+    logging.warning(
+        'Difference between first and last stream start/end times was %s and '
+        '%s, which is more than %s of the benchmark time %s.',
```

Comment:
"the benchmark time" should be "all_streams_duration + start_gap + stop_gap", right?